### DISCUSSION TITLE
[OP Stack Deployment] OP-geth chain ID

### DISCUSSION THREAD

**Discussion by zhaozaao** [ethereum-optimism/developers#552] [Deploying the OP Stack] [answered]
### Did you check the documentation?

- [X] Yes

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

I have been trying to deploy an OP stack rollup. I have succeeded in the past, but today when I try to restart my rollup, something wrong happens with starting the proposer(something is wrong with the l2oo I suppose, as I haven't restarted the rollup for about two weeks), so I try to go over the whole tutorials again and redeploy another rollup. But it seems that the op-geth is always linking to the main net which the chain ID is 1, I have no idea where to change this. The op-node genesis file only have the option to change the L2 chain ID I suppose?  And this causes the problem of starting op-node, because the log says incorrect L2 RPC chain ID.

Links to the tutorial： https://docs.optimism.io/builders/chain-operators/tutorials/create-l2-rollup
OP-node version：0.10.14
Runtime Configuration：
./bin/op-node \
  --l2=http://localhost:8551 \
  --l2.jwt-secret=./jwt.txt \
  --sequencer.enabled \
  --sequencer.l1-confs=5 \
  --verifier.l1-confs=4 \
  --rollup.config=./rollup.json \
  --rpc.addr=0.0.0.0 \
  --p2p.disable \
  --rpc.enable-admin \
  --p2p.sequencer.key=$GS_SEQUENCER_PRIVATE_KEY \
  --l1=$L1_RPC_URL \
  --l1.rpckind=$L1_RPC_KIND

OP-geth version：1.14.7 as in op-geth/parms/verision.go
runtime configuration:
./build/bin/geth \
  --datadir ./datadir \
  --http \
  --http.corsdomain="*" \
  --http.vhosts="*" \
  --http.addr=0.0.0.0 \
  --http.api=web3,debug,eth,txpool,net,engine \
  --ws \
  --ws.addr=0.0.0.0 \
  --ws.port=8546 \
  --ws.origins="*" \
  --ws.api=debug,eth,txpool,net,engine \
  --syncmode=full \
  --gcmode=archive \
  --nodiscover \
  --maxpeers=0 \
  --networkid=42069 \
  --authrpc.vhosts="*" \
  --authrpc.addr=0.0.0.0 \
  --authrpc.port=8551 \
  --authrpc.jwtsecret=./jwt.txt \
  --rollup.disabletxpoolgossip=true

### Logs

op-geth:
```
INFO [08-11|19:29:53.696] Maximum peer count                       ETH=0 total=0
INFO [08-11|19:29:53.700] Enabling recording of key preimages since archive mode is used
WARN [08-11|19:29:53.700] Disabled transaction unindexing for archive node
WARN [08-11|19:29:53.700] Forcing hash state-scheme for archive mode
INFO [08-11|19:29:53.701] Set global gas cap                       cap=50,000,000
INFO [08-11|19:29:53.702] Initializing the KZG library             backend=gokzg
INFO [08-11|19:29:53.721] Allocated trie memory caches             clean=307.00MiB dirty=0.00B
INFO [08-11|19:29:53.721] Using pebble as the backing database
INFO [08-11|19:29:53.721] Allocated cache and file handles         database=C:\Users\86186\op-geth\datadir\geth\chaindata cache=512.00MiB handles=8192
INFO [08-11|19:29:53.742] Opened ancient database                  database=C:\Users\86186\op-geth\datadir\geth\chaindata\ancient\chain readonly=false
INFO [08-11|19:29:53.742] State scheme set by user                 scheme=hash
INFO [08-11|19:29:53.750]
INFO [08-11|19:29:53.750] ---------------------------------------------------------------------------------------------------------------------------------------------------------
INFO [08-11|19:29:53.750] Chain ID:  1 (mainnet)
INFO [08-11|19:29:53.750] Consensus: Beacon (proof-of-stake), merged from Ethash (proof-of-work)
INFO [08-11|19:29:53.750]
INFO [08-11|19:29:53.750] Pre-Merge hard forks (block based):
INFO [08-11|19:29:53.750]  - Homestead:                   #1150000  (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/homestead.md)
INFO [08-11|19:29:53.750]  - DAO Fork:                    #1920000  (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/dao-fork.md)
INFO [08-11|19:29:53.750]  - Tangerine Whistle (EIP 150): #2463000  (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/tangerine-whistle.md)
INFO [08-11|19:29:53.750]  - Spurious Dragon/1 (EIP 155): #2675000  (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/spurious-dragon.md)
INFO [08-11|19:29:53.750]  - Spurious Dragon/2 (EIP 158): #2675000  (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/spurious-dragon.md)
INFO [08-11|19:29:53.750]  - Byzantium:                   #4370000  (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/byzantium.md)
INFO [08-11|19:29:53.750]  - Constantinople:              #7280000  (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/constantinople.md)
INFO [08-11|19:29:53.750]  - Petersburg:                  #7280000  (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/petersburg.md)
INFO [08-11|19:29:53.750]  - Istanbul:                    #9069000  (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/istanbul.md)
INFO [08-11|19:29:53.750]  - Muir Glacier:                #9200000  (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/muir-glacier.md)
INFO [08-11|19:29:53.750]  - Berlin:                      #12244000 (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/berlin.md)
INFO [08-11|19:29:53.750]  - London:                      #12965000 (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/london.md)
INFO [08-11|19:29:53.750]  - Arrow Glacier:               #13773000 (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/arrow-glacier.md)
INFO [08-11|19:29:53.750]  - Gray Glacier:                #15050000 (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/gray-glacier.md)
INFO [08-11|19:29:53.750]
INFO [08-11|19:29:53.750] Merge configured:
INFO [08-11|19:29:53.750]  - Hard-fork specification:    https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/paris.md
INFO [08-11|19:29:53.750]  - Network known to be merged: true
INFO [08-11|19:29:53.750]  - Total terminal difficulty:  58750000000000000000000
INFO [08-11|19:29:53.750]
INFO [08-11|19:29:53.750] Post-Merge hard forks (timestamp based):
INFO [08-11|19:29:53.750]  - Shanghai:                    @1681338455 (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/shanghai.md)
INFO [08-11|19:29:53.750]  - Cancun:                      @1710338135 (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/cancun.md)
INFO [08-11|19:29:53.750]
INFO [08-11|19:29:53.750] ---------------------------------------------------------------------------------------------------------------------------------------------------------
INFO [08-11|19:29:53.750]
INFO [08-11|19:29:53.750] Loaded most recent local block           number=0 hash=d4e567..cb8fa3 td=17,179,869,184 age=55y4mo3w
INFO [08-11|19:29:53.750] Initialized transaction indexer          range="entire chain"
INFO [08-11|19:29:53.750] Initialising Ethereum protocol           network=42069 dbversion=8
INFO [08-11|19:29:53.751] Loaded local transaction journal         transactions=0 dropped=0
INFO [08-11|19:29:53.774] Gasprice oracle is ignoring threshold set threshold=2
WARN [08-11|19:29:53.777] Engine API enabled                       protocol=eth
INFO [08-11|19:29:53.777] Starting peer-to-peer node               instance=Geth/v1.101407.0-rc.1-ee83530a-20240809/windows-amd64/go1.22.5
INFO [08-11|19:29:53.798] New local node record                    seq=1,723,400,882,879 id=c2d7561294d3374f ip=127.0.0.1 udp=0 tcp=30303
INFO [08-11|19:29:53.798] Started P2P networking                   self="enode://e7db6455c75b773334ba32620d37aa1e88d5475ce77d742950b413b612309ad917ee79b0e95d8e27bd45400ab0501e384a60cc6f966c0d26c2e3a9a45ea9cdad@127.0.0.1:30303?discport=0"
INFO [08-11|19:29:53.799] IPC endpoint opened                      url=\\.\pipe\geth.ipc
INFO [08-11|19:29:53.800] Loaded JWT secret file                   path=jwt.txt crc32=0x1cb0a7d
INFO [08-11|19:29:53.800] HTTP server started                      endpoint=[::]:8545 auth=false prefix= cors=* vhosts=*
INFO [08-11|19:29:53.800] WebSocket enabled                        url=ws://[::]:8546
INFO [08-11|19:29:53.800] WebSocket enabled                        url=ws://[::]:8551
INFO [08-11|19:29:53.800] HTTP server started                      endpoint=[::]:8551 auth=true  prefix= cors=localhost vhosts=*
WARN [08-11|19:30:28.804] Post-merge network, but no beacon client seen. Please launch one to follow the chain!
WARN [08-11|19:35:29.279] Post-merge network, but no beacon client seen. Please launch one to follow the chain!
```

op-node:
```
t=2024-08-11T19:30:06+0100 lvl=info msg="Not opted in to ProtocolVersions signal loading, disabling ProtocolVersions contract now."
t=2024-08-11T19:30:06+0100 lvl=info msg="No persisted sequencer state loaded"
t=2024-08-11T19:30:06+0100 lvl=info msg="Rollup Config"                       l2_chain_id=42069 l2_network="unknown L2" l1_chain_id=11,155,111 l1_network=sepolia l2_start_time=1,723,398,984 l2_block_hash=0x5f958accfbb14ded3e321aaa6e87cd5285bcc1a6074edd73cf029d5e9316ae15 l2_block_number=0 l1_block_hash=0x64cc4bb9745cae8e02b1b1de68651ad746c2ddaccec2a94a50776c777fc226e0 l1_block_number=6,480,988 regolith_time="@ genesis" canyon_time="@ genesis" delta_time="(not configured)" ecotone_time="(not configured)" fjord_time="(not configured)" interop_time="(not configured)"
t=2024-08-11T19:30:06+0100 lvl=info msg="Initializing rollup node"            version=v0.0.0-a06cae81-1705510057
t=2024-08-11T19:30:07+0100 lvl=warn msg="No beacon endpoint configured. Configuration is mandatory for the Ecotone upgrade"
t=2024-08-11T19:30:07+0100 lvl=eror msg="Error initializing the rollup node"  err="failed to init L2: incorrect L2 RPC chain id 1, expected 42069"
t=2024-08-11T19:30:07+0100 lvl=crit msg="Application failed"                  message="failed to setup: unable to create the rollup node: failed to init L2: incorrect L2 RPC chain id 1, expected 42069"
```

### Additional Information

When I create the genesis file following the tutorials, it says head block is not reachable. And there is also something wrong with this error:sers\86186\op-geth\datadir\geth\chaindata\ancient\chain readonly=false
Fatal: Failed to register the Ethereum service: incompatible state scheme, stored: path, provided: hash
So I just deleted the chain data folder. I think my database must have been spoiled, but I have no idea how to totally get rid of the old one. I deleted the whole op-geth and reinstall it, but it still says INFO [08-11|19:27:16.075] Opened ancient database                  database=C:\Users\86186\op-geth\datadir\geth\lightchaindata\ancient\state readonly=false. How can I clear all the ancient database and restart?

**Comment by hash1go** [comment]
Hello!

To help clarify your situation, could you please specify:

1. Which L1 chain are you trying to deploy your rollup on? (Ethereum mainnet or Sepolia testnet?)
2. What chain ID are you aiming to use for your L2 rollup?

**Reply by zhaozaao** [reply]
Hello！ I am trying to deploy on the sepolia testnet. The L2 chain Id is the default 42069.

**Comment by Chomtana** [comment]
It should be about the genesis file generation where it says head block is not reachable

Do you have a full log on this?

**Reply by zhaozaao** [reply]
Hello! This is the full log:
$ build/bin/geth init --datadir=datadir genesis.json
INFO [08-13|16:42:26.802] Maximum peer count                       ETH=50 total=50
INFO [08-13|16:42:26.809] Set global gas cap                       cap=50,000,000
INFO [08-13|16:42:26.810] Initializing the KZG library             backend=gokzg
INFO [08-13|16:42:26.829] Defaulting to pebble as the backing database
INFO [08-13|16:42:26.829] Allocated cache and file handles         database=C:\Users\86186\op-geth\datadir\geth\chaindata cache=16.00MiB handles=16
INFO [08-13|16:42:26.873] Opened ancient database                  database=C:\Users\86186\op-geth\datadir\geth\chaindata\ancient\chain readonly=false
INFO [08-13|16:42:26.873] State schema set to default              scheme=path
ERROR[08-13|16:42:26.873] Head block is not reachable
INFO [08-13|16:42:26.907] Opened ancient database                  database=C:\Users\86186\op-geth\datadir\geth\chaindata\ancient\state readonly=false
INFO [08-13|16:42:26.907] Writing custom genesis block
INFO [08-13|16:42:27.240] Successfully wrote genesis state         database=chaindata hash=93ab04..fb5839
INFO [08-13|16:42:27.241] Defaulting to pebble as the backing database
INFO [08-13|16:42:27.241] Allocated cache and file handles         database=C:\Users\86186\op-geth\datadir\geth\lightchaindata cache=16.00MiB handles=16
INFO [08-13|16:42:27.285] Opened ancient database                  database=C:\Users\86186\op-geth\datadir\geth\lightchaindata\ancient\chain readonly=false
INFO [08-13|16:42:27.285] State schema set to default              scheme=path
ERROR[08-13|16:42:27.285] Head block is not reachable
INFO [08-13|16:42:27.314] Opened ancient database                  database=C:\Users\86186\op-geth\datadir\geth\lightchaindata\ancient\state readonly=false
INFO [08-13|16:42:27.314] Writing custom genesis block
INFO [08-13|16:42:27.624] Successfully wrote genesis state         database=lightchaindata hash=93ab04..fb5839

**Reply by zhaozaao** [reply]
If I delete the datadir/geth folder, op-geth can start normally, but op-node still can not because of the chain ID

**Comment by zhaozaao** [comment]
I have used a alchemy RPC linked to the Sepolia, why would the op-geth connected to the mainnet?

**Comment by sbvegan** [comment]
Hey @zhaozaao, can you please edit the original post with the following information:

- op-node version and runtime configuration
- op-geth version and runtime configuration
- link to the tutorial section you followed to generate the genesis.json

**Reply by zhaozaao** [reply]
Hi @sbvegan , I have edited the original post.

**Comment by zhaozaao** [comment, marked as answer]
I solve this problem by using state schema as hash when creating the genesis file.
$ ./build/bin/geth init --datadir=datadir --state.scheme hash genesis.json
